src/argilla/server/apis/v1/handlers/datasets.py

```diff
@@ -91,6 +97,147 @@ async def _get_dataset(
+) -> List["MetadataFilter"]:
+    try:
+        metadata_filters = []
+        for metadata_param in parsed_metadata:
+            metadata_property = await datasets.get_metadata_property_by_name_and_dataset_id(
+                db, name=metadata_param.name, dataset_id=dataset.id
+            )
+            if metadata_property is None:
+                continue  # won't fail on unknown metadata filter name
+
+            if metadata_property.type == MetadataPropertyType.terms:
+                metadata_filter_class = TermsMetadataFilter
+            elif metadata_property.type == MetadataPropertyType.integer:
+                metadata_filter_class = IntegerMetadataFilter
+            elif metadata_property.type == MetadataPropertyType.float:
+                metadata_filter_class = FloatMetadataFilter
+            else:
+                raise ValueError(f"Not found filter for type {metadata_property.type}")
+
+            metadata_filters.append(metadata_filter_class.from_string(metadata_property, metadata_param.value))
+    except ValueError as ex:
+        raise HTTPException(
+            status_code=status.HTTP_422_UNPROCESSABLE_ENTITY, detail=f"Cannot parse provided metadata filters: {ex}"
+        )
+    return metadata_filters
+
+
+async def _build_response_status_filter_for_search(
+    response_statuses: Optional[List[ResponseStatusFilter]] = None, user: Optional[User] = None
+) -> Optional[UserResponseStatusFilter]:
+    user_response_status_filter = None
+
+    if response_statuses:
+        # TODO(@frascuchon): user response and status responses should be split into different filter types
+        user_response_status_filter = UserResponseStatusFilter(user=user, statuses=response_statuses)
+
+    return user_response_status_filter
+
+
+_RECORD_SORT_FIELD_VALUES = tuple(field.value for field in RecordSortField)
+_VALID_SORT_VALUES = tuple(sort.value for sort in SortOrder)
+
+
+async def _build_sort_by(
+    db: "AsyncSession", dataset: Dataset, sort_by_query_param: Optional[Dict[str, str]] = None
+) -> Union[List[SortBy], None]:
+    if sort_by_query_param is None:
+        return None
+
+    sorts_by = []
+    for sort_field, sort_order in sort_by_query_param.items():
+        if sort_field not in _RECORD_SORT_FIELD_VALUES:
+            metadata_property = await datasets.get_metadata_property_by_name_and_dataset_id(
+                db, name=sort_field, dataset_id=dataset.id
+            )
+            if not metadata_property:
+                raise HTTPException(
+                    status_code=status.HTTP_422_UNPROCESSABLE_ENTITY,
+                    detail=(
+                        f"Provided metadata property in 'sort_by' query param '{sort_field}' not found in dataset with"
+                        f" '{dataset.id}'."
+                    ),
+                )
+            field = metadata_property
+        else:
+            field = sort_field
```

Comment:
I think inverting the if-else logic here, the code will be more readable (long clause in the `else` part):
```python
if sort_field in _RECORD_SORT_FIELD_VALUES:
   field = sort_field
else:
   long code block here
```